src/exposition/http.rs

```diff
@@ -107,6 +122,58 @@ mod handlers {
                         }
                     }
                 }
+                if config.prometheus().histograms() {
+                    if let Some(snapshot) = snapshots.previous.get(metric.name()) {
+                        // downsample the snapshot if necessary
+                        let downsampled: Option<Snapshot> =
+                            if config.prometheus().histogram_downsampling_factor() > 0 {
```

Comment:
this comparison should also consider the original grouping factor not just if down-sampling factor is set

Reply:
Why is that? The downsampling factor is how much we reduce the grouping power. We check that the downsample factor is valid (in relation to the grouping power) when we load the config.

Reply:
I see. `downsample` already checks that. But if the config is set to be higher than the original histogram, it would return an error during exposition. Seems something that should be verified during initialization if we don't want to silently swallow that misconfiguration?